server/src/configs/authentication.js

```diff
@@ -23,18 +26,27 @@ Authentication.prototype.start = () => {
     {
       usernameField: 'email',
       passwordField: 'password',
+      passReqToCallback: true,
     },
-    async (email, password, done) => {
-      // TODO: Checks for sign up (Waiting for Data models to be added)
-      /**
-       * Steps for sign up.
-       * 1.check if the user already exists in the database and return error in done callback.
-       * 2. If user does not exists,
-       *        -create a user in the database and return the created user object.
-       */
-
-      const user = { email };
-      return done(null, user);
+    async (req, email, password, done) => {
+      try {
+        const emailAlreadyExists = await userExists(email);
```

Comment:
Instead of individually checking username and email , you can check using single mongo query with or condition.